python/pyspark/sql/tests/test_functions.py

```diff
@@ -1032,9 +1032,11 @@ def check_assert_true(self, tpe):
         )
 
         with self.assertRaisesRegex(tpe, "too big"):
-            df.select(F.assert_true(df.id < 2, "too big")).toDF("val").collect()
+            df.select(F.assert_true(df.id < 2, "[USER_RAISED_EXCEPTION] too big")).toDF(
+                "val"
+            ).collect()
 
-        with self.assertRaisesRegex(tpe, "2000000"):
+        with self.assertRaisesRegex(tpe, "[USER_RAISED_EXCEPTION] 2000000.0"):
```

Comment:
```suggestion
        with self.assertRaisesRegex(tpe, "\[USER_RAISED_EXCEPTION\] 2000000.0"):
```

Maybe this should work? `[` is treated as a special character for regex, so we should escape it.